### ISSUE TITLE
(CDK CLI): migrate doesn't produce correct IAM policies

### ISSUE THREAD

**Issue by mbonig** [aws/aws-cdk#28014]
### Describe the bug

When importing this template:
https://github.com/awslabs/aws-cloudformation-templates/blob/master/aws/solutions/LambdaStaticIP/lambda-static.cfn.yaml

The IAM role has bad casing on the policy:

```
    const lambdaRole = new iam.CfnRole(this, 'LambdaRole', {
      assumeRolePolicyDocument: {
        version: '2012-10-17',
        statement: [
          {
            effect: 'Allow',
            principal: {
              service: 'lambda.amazonaws.com',
            },
            action: 'sts:AssumeRole',
          },
        ],
      },
      managedPolicyArns: [
        'arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole',
      ],
      path: '/',
      policies: [
        {
          policyName: 'EC2Permissions',
          policyDocument: {
            version: '2012-10-17',
            statement: [
              {
                effect: 'Allow',
                action: 'ec2:DescribeNetworkInterfaces',
                resource: '*',
              },
              {
                effect: 'Allow',
                action: 'ec2:CreateNetworkInterface',
                resource: '*',
              },
              {
                effect: 'Allow',
                action: 'ec2:DeleteNetworkInterface',
                resource: '*',
              },
              {
                effect: 'Allow',
                action: 'ec2:DescribeInstances',
                resource: '*',
              },
              {
                effect: 'Allow',
                action: 'ec2:AttachNetworkInterface',
                resource: '*',
              },
            ],
          },
        },
      ],
    });

```

### Expected Behavior

I expected the IAM policy documents to have the correct casing and be deployable.

### Current Behavior

lowercase properties are incorrect for these IAM policy statements

### Reproduction Steps

run cdk migrate on the linked template. Try to deploy, it will fail.

### CDK CLI Version

2.106.0 (build 0d1a333)

### Node.js Version

v20.9.0

### OS

Mac OS 14.1.1

### Language

TypeScript

**Comment by khushail** [comment]
thanks @mbonig for reporting this.

**Comment by TheRealAmazonKendra** [comment]
The fix has now been released as of version 2.119.0.
